src/java.base/share/classes/java/security/Provider.java

```diff
@@ -1372,6 +1381,12 @@ protected void putService(Service s) {
             throw new IllegalArgumentException
                     ("service.getProvider() must match this Provider object");
         }
+        if (SecurityRestrictConfigurator.enableSecurityRestrict()) {
+            // If security restrict is not allow, then return without register
+            if (!SecurityRestrictProperties.getInstance().isConstraintsAllow(s)) {
```

Comment:
I think we can find better names for the classes and methods so this reads more clearly. I suggest:
* SecurityRestrictConfigurator -> RestrictedSecurityConfigurator
* SecurityRestrictProperties -> RestrictedSecurityProperties
* enableSecurityRestrict -> isEnabled
* isConstraintsAllow -> isServiceAllowed

The comment should have a period at the end of the sentence, 

> // If security restrictions do not allow the service, then return without registering.

A better name for `s` would be `service`.

Reply:
@keithc-ca 
Thanks for the review. I updated the codes according to your review and also make other changes to make methods read more clearly. Please help to review.

@alon-sh  fyi